src/Handler.ts

```diff
@@ -339,20 +347,77 @@ class Handler extends Eventful {
         const out = new HoveredResult(x, y);
 
         for (let i = list.length - 1; i >= 0; i--) {
+            const el = list[i];
             let hoverCheckResult;
-            if (list[i] !== exclude
+            if (el !== exclude
                 // getDisplayList may include ignored item in VML mode
-                && !list[i].ignore
-                && (hoverCheckResult = isHover(list[i], x, y))
+                && !el.ignore
+                && (hoverCheckResult = isHover(el, x, y))
             ) {
-                !out.topTarget && (out.topTarget = list[i]);
+                !out.topTarget && (out.topTarget = el);
                 if (hoverCheckResult !== SILENT) {
-                    out.target = list[i];
+                    out.target = el;
                     break;
                 }
             }
         }
 
+        if (this._pointerSize && !out.target) {
+            /**
+             * If no element at pointer position, check intersection with
+             * elements with pointer enlarged by target size.
+             */
+            const candidates = [];
+            const pointerSize = this._pointerSize;
+            const targetSizeHalf = pointerSize / 2;
+            const pointerRect = new BoundingRect(x - targetSizeHalf, y - targetSizeHalf, pointerSize, pointerSize);
+
+            for (let i = list.length - 1; i >= 0; i--) {
+                const el = list[i];
+                if (el !== exclude
+                    && !el.ignore
+                    && !el.ignoreCoarsePointer
+                    // If an element ignores, its textContent should also ignore.
+                    // TSpan's parent is not a Group but a ZRText.
+                    // See Text.js _getOrCreateChild
+                    && (!el.parent || !(el.parent as any).ignoreCoarsePointer)
+                ) {
+                    tmpRect.copy(el.getBoundingRect());
+                    if (el.transform) {
+                        tmpRect.applyTransform(el.transform);
+                    }
+                    if (tmpRect.intersect(pointerRect)) {
+                        candidates.push(el);
+                    }
+                }
+            }
+
+            /**
+             * If there are elements whose bounding boxes are near the pointer,
+             * use the most top one that intersects with the enlarged pointer.
+             */
+            if (candidates.length) {
+                const rStep = 4;
+                const thetaStep = Math.PI / 12;
+                for (let r = 0; r < targetSizeHalf && !out.target; r += rStep) {
+                    for (let theta = 0; theta < Math.PI * 2 && !out.target; theta += thetaStep) {
```

Comment:
Because in the inner loop it will break after assigning to `out.target`. There is no need to check `out.target` here

Reply:
I think this is still necessary because `break` can only break one level of inner loop. Can you double check if I understand it right?

Reply:
Why not directly return the `out` target when it's found? We can move the judgment from two `for` loops to the back of `setHoverTarget` function.

```js
for (outer loop) {
  for (inner loop) {
    // ....
    setHoverTarget(xxxx);
    if (out.target) {
      return out;
    }
  }
}
```

Reply:
Good idea